nemo/lightning/io/pl.py

```diff
@@ -118,7 +139,7 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_optio
                 f" storage_options, but {storage_options=} was provided."
                 f" Ignoring given storage_options"
             )
-        checkpoint_dir = ckpt_to_dir(path)
+        checkpoint_dir = ckpt_to_weights_subdir(path, is_saving=True)
```

Comment:
For the save part, can we include `assert checkpoint_dir.parts[-1] == WEIGHTS_PATH, checkpoint_dir` ?